nuitka/build/static_src/CompiledCodeHelpers.c

```diff
@@ -541,6 +541,42 @@ PyObject *BUILTIN_LEN(PyObject *value) {
     return PyInt_FromSsize_t(res);
 }
 
+PyObject *BUILTIN_ANY(PyObject *value) {
+    CHECK_OBJECT(value);
+
+    PyObject *it = PyObject_GetIter(value);
+    PyObject *(*iternextfunc)(PyObject *) = *Py_TYPE(it)->tp_iternext;;
```

Comment:
I think the Python header object.h defines `iternextfunc`, and I recommend using it like `iternextfunc iternext = Py_TYPE(it)->tp_iternext;` (btw there is two `;` there in your code).

Reply:
And besides, in similar vein to my next line comment, move this done to where it is actually used, that would be right before the loop.

Here you call `GetIter` and then assign around some pointers, and only then use the value. The C compiler will be unsure if it doesn't have to segfault before you return and actually do it early, only to not use it. It definitely is easier to read, although it is probably more distant to the CPython code (does it do it that way, I doubt though), but that does not matter, just keep variable declarations and assignments together, and as late as possible please.

Reply:
> The C compiler will be unsure if it doesn't have to segfault before you return and actually do it early, only to not use it.

Okay sure I will keep variable declarations and assignments together but I don't get it how declaring at once is going to give me a segfault?

Reply:
I didn't respond to this yet.

Well, what you do is basically in Python equivalent to this:

```
x = f()
a = y.z
if x is None:
   return None
```

Even if the C compiler might optimize the assignment to `a` away and it will, why ask the reader to also read it like that?

Reply:
Okay. So it's better to keep related expressions together for better readability.